splice_machine/src/test/java/com/splicemachine/derby/impl/sql/compile/UpdateFromSubqueryIT.java

```diff
@@ -491,6 +494,91 @@ public void testNestedFromSubquery() throws Exception {
         try (ResultSet rs = spliceClassWatcher.executeQuery("select * from t1")) {
             assertEquals(expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
         }
+
+        // UPDATE ... FROM, single subquery
+        // where clause is part of the update statement
+        spliceClassWatcher.executeUpdate(format("update t1 set b1 = b2" +
+                " from (select a2, b2 from t2 --splice-properties joinStrategy=%s,useSpark=%s\n" +
+                " cross join (select a3 from t3)) where a1=a2", this.joinStrategy, this.useSparkString));
+
+        try (ResultSet rs = spliceClassWatcher.executeQuery("select * from t1")) {
+            assertEquals(expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
+        }
+
+        // UPDATE ... FROM, from list
+        // where clause is part of the update statement
+        spliceClassWatcher.executeUpdate(format("update t1 set b1 = b2" +
```

Comment:
Do we support updating multiple columns with this syntax? Can you add a test if so?

Reply:
Yes, updating multiple columns using this syntax is supported. I added a new test for this case. Basically, it's an existing test but the query is rewritten using the new syntax.